pkgs/tools/typesetting/tex/texlive/default.nix

```diff
@@ -502,8 +502,26 @@ let
          fods = filter (p: isDerivation p && p.tlType != "bin") all;
       in builtins.all (p: assertMsg (p ? outputHash) "The TeX Live package '${p.pname + lib.optionalString (p.tlType != "run") ("." + p.tlType)}' does not have a fixed output hash. Please read UPGRADING.md on how to build a new 'fixed-hashes.nix'.") fods));
 
+  # texlive.combine compatibility layer:
+  # export packages as `texlive.pname = { pkgs = [ ... ]; };`
+  combineCompatLayer = attrs:
+    lib.mapAttrs (n: p: {
+      pkgs = p.pkgs;
+    }) attrs.__pkgs // attrs;
+
+  __buildEnv = import ./buildenv.nix {
+    self = { inherit bin; __pkgs = tl; };
+    ghostscript = ghostscript_headless;
+    inherit lib buildEnv libfaketime makeFontsConf makeWrapper runCommand
+      writeText bash perl coreutils gawk gnugrep gnused;
+  };
```

Comment:
I forgot that `buildEnv` is a *derivation* in Lua, Perl, etc. I presume because it splices correcly.

Reply:
It's now a derivation on which you call `.override`.

Reply:
I understand now why `buildEnv` is a derivation in Lua etc, and it has to do with how the packages are exported at the top level (`luaPackages`, `lua54Packages`, `lua5_4` etc). Since `texlive` is a recurse-for-derivation set, `buildEnv` must be a function or it will pop up in search, which of course we do not want. This can change later if we go for `texlive2022`, `texlive2022Full`, `texlivePackages` etc.